trade-finance-manager-ui/server/controllers/case/cancellation/effective-from-date.controller.ts

```diff
@@ -17,19 +20,43 @@ export type PostEffectiveFromDateRequest = CustomExpressRequest<{
  * @param req - The express request
  * @param res - The express response
  */
-export const getEffectiveFromDate = (req: GetEffectiveFromDateRequest, res: Response) => {
+export const getEffectiveFromDate = async (req: GetEffectiveFromDateRequest, res: Response) => {
   const { _id } = req.params;
-  const { user } = asUserSession(req.session);
+  const { user, userToken } = asUserSession(req.session);
 
-  // TODO DTFS2-7417: Check deal type allows for cancellation and deal hasn't already been cancelled
+  try {
+    const deal = await api.getDeal(_id, userToken);
 
-  const effectiveFromDateViewModel: EffectiveFromDateViewModel = {
-    activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.ALL_DEALS,
-    user,
-    ukefDealId: '0040613574', // TODO DTFS2-7417: get values from database
-    dealId: _id,
-  };
-  return res.render('case/cancellation/effective-from-date.njk', effectiveFromDateViewModel);
+    if (!deal || 'status' in deal) {
+      return res.redirect('/not-found');
+    }
+
+    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
+      return res.redirect(`/case/${_id}/deal`);
+    }
+
+    const cancellation = await api.getDealCancellation(_id, userToken);
+
+    const previouslyEnteredEffectiveFromDate = cancellation?.effectiveFrom && new Date(cancellation.effectiveFrom);
+
+    const day = previouslyEnteredEffectiveFromDate ? format(previouslyEnteredEffectiveFromDate, 'd') : '';
+    const month = previouslyEnteredEffectiveFromDate ? format(previouslyEnteredEffectiveFromDate, 'M') : '';
+    const year = previouslyEnteredEffectiveFromDate ? format(previouslyEnteredEffectiveFromDate, 'yyyy') : '';
```

Comment:
Not a blocker for this PR - we should probably create a helper for this, I can see such logic is already in place in 2/3 other files, it could become something like this in individual controllers and unit tests 😎 

```ts
const effectiveFromDateViewModel: EffectiveFromDateViewModel = {
  activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.ALL_DEALS,
  user,
  ukefDealId: deal.dealSnapshot.details.ukefDealId,
  dealId: _id,
  ...formattedDate(previouslyEnteredEffectiveFromDate),   // <<<< { day, month, year }
};
```

Reply:
Good point! I'll merge this one as is to keep things moving, but will bear in mind going forwards.